### DISCUSSION TITLE
Migrate to VitePress

### DISCUSSION THREAD

**Discussion by melishev** [socketio/socket.io#5180] [Ideas]
Congratulations on your successful transition to a monorepository!

There is a proposal to move the project to VitePress, replacing Docusaurus, which the project is currently using. I see that a lot of the client code in the examples is written in Vue.js and I think that VitePress would be a good addition to the mix.
If @darrachequesne is willing to make that change, I'd be happy to handle the transition!

**Comment by darrachequesne** [comment]
> Congratulations on your successful transition to a monorepository!

Thanks :heart: 

I'm not really familiar with VitePress, what would be the benefits over Docusaurus?

**Reply by melishev** [reply]
I don't think I can give you an exact comparison between Docusaurus and VitePress since I haven't worked much with the former. As far as I understand it now, it's like comparing Webpack and Vite.
My personal opinion is that the socket.io project could use a visual upgrade to VitePress.
If you've thought about it or would like to bring this project into Vite and VitePress, I'd love to help with that.

**Reply by melishev** [reply]
Check out VitePress, they've made a very informative but not gigantic [documentation](https://vitepress.dev/guide/what-is-vitepress) that is a pleasure to read.

Maybe after reading it, you'll give it a go!

**Reply by melishev** [reply]
Also as part of this task or the next, you could move the repository itself with documentation to a monorepository.
What do you think @darrachequesne ?
